### PR TITLE
Adding Pointed Categories 

### PR DESCRIPTION
**Pull request by SpencerWoolfson**
I initially set this up using Hofmann and Streicher's method. I have fixed it to use the correct setup.

### CODE REVIEW

File: GroupoidModel/Groupoids.lean [hunk 1]
```diff
@@ -152,154 +153,91 @@ def functorial {C D : Grpd.{v₁,u₁}} (F : C ⟶ D) (G : D ⥤ Grpd.{v₂,u₂
...
-section HSexp
 
-/-
-In this section we go through section 4 of Hofmann and Streicher's original paper
--/
+section PointedCategorys
+
+structure PointedCategory.{w,z} extends Pointed.{z} where
+  cat : Category.{w} X
+
```
Comment (sinhp) [inline GroupoidModel/Groupoids.lean:161]: I think it is more natural to define `PointedCategory` as a category with a point. Same with pointed groupoids.

Comment (sinhp) [reply GroupoidModel/Groupoids.lean:161]:
```suggestion
/-- The structure of pointed groupoids. -/
class PointedGroupoid (X : Type*) extends Groupoid X where
  pt : X
```

Comment (sinhp) [reply GroupoidModel/Groupoids.lean:161]: This way you can define `Hom C D` as a functor which repsects the point:
```suggestion
/-- A morphism of pointed groupoids is a functor that preserves the distinguished point. -/
@[ext]
protected structure Hom (C D : Type*) [PointedGroupoid C] [PointedGroupoid D] where
  /-- The underlying map of pointed types. -/
  toFunc : C ⥤ D
  toFunc_point : toFunc.obj PointedGroupoid.pt = PointedGroupoid.pt
```

File: GroupoidModel/Groupoids.lean [hunk 2]
```diff
@@ -152,154 +153,91 @@ def functorial {C D : Grpd.{v₁,u₁}} (F : C ⟶ D) (G : D ⥤ Grpd.{v₂,u₂
...
--- This is a Covariant Functor that takes a Groupoid Γ to Ty Γ
+def of.{w,z} {X : Type z} (point : X)[grpd : Groupoid.{w} X]: PointedGroupoid :=
+  ⟨⟨X,point⟩,grpd⟩
+
+instance largeCategory : LargeCategory PointedGroupoid where
+  Hom P Q := PointedCategory.Hom P Q
+  id P := PointedCategory.Hom.id P
+  comp f g := PointedCategory.Hom.comp f g
+
+def forgetPoint.{w,z} : PointedGroupoid.{w,z} ⥤ Grpd.{w,z} where
```
Comment (sinhp) [inline GroupoidModel/Groupoids.lean:213]: Definitions need doc string explaining what is being defined: for the style see https://leanprover-community.github.io/contribute/style.html

**Review by sinhp — CHANGES_REQUESTED** [review]
docstrings for definitions and possibly (?) changes to`PointedGroupoid`/`PointedCategory`

File: GroupoidModel/Groupoids.lean (hunk 1 above)
Comment (SpencerWoolfson) [reply GroupoidModel/Groupoids.lean:161]: Thanks, I will make these changes.

**Comment by sinhp** [comment]
LGTM

**Pull request merged**